src/providers/mssql/qgsmssqlprovider.cpp

```diff
@@ -358,6 +359,29 @@ bool QgsMssqlProvider::execLogged( QSqlQuery &qry, const QString &sql, const QSt
   return res;
 }
 
+bool QgsMssqlProvider::execPreparedLogged( QSqlQuery &qry, const QString &queryOrigin ) const
```

Comment:
Is it possible to separate the PR in two. One for the polygon fix and another for the MSSQL improvement?

Reply:
Everything is possible, but takes time, since I am doing this as a volunteer in my spare time for the glory of the project I think we should be tolerant and weight the pros and cons. 

I don't guarantee I'll find the motivation.